src/api/firebase.js

```diff
@@ -133,3 +125,99 @@ export async function matchToken(listId) {
 	const jointListTokenSnapshot = await getDocs(jointListTokenQuery);
 	return jointListTokenSnapshot;
 }
+
+export function comparePurchaseUrgency(items) {
+	const overdueList = [];
+	const soonList = [];
+	const kindOfSoonList = [];
+	const notSoonList = [];
+	const purchasedList = [];
+	const inactiveList = [];
+
+	items.forEach((item) => {
+		const differenceInDays = item.dateLastPurchased
+			? getDaysBetweenDates(item.dateLastPurchased, item.dateNextPurchased)
+			: getDaysBetweenDates(item.dateCreated, item.dateNextPurchased);
+
+		if (differenceInDays >= 60) {
+			item.isInactive = true;
+			item.urgency = 'Inactive';
+		} else {
+			const differenceTillNextPurchase = getDaysBetweenDates(
+				item.dateNextPurchased,
+			);
+
+			item.isInactive = false;
+			item.days = differenceTillNextPurchase;
+
+			// switch (true) {
+			// 	case differenceTillNextPurchase < 0:
+			// 		item.urgency = 'Overdue';
+			// 		break;
+			// 	case differenceTillNextPurchase <= 7:
+			// 		item.urgency = 'Soon';
+			// 		break;
+			// 	case differenceTillNextPurchase > 7 && differenceTillNextPurchase < 30:
+			// 		item.urgency = 'Kind of soon';
+			// 		break;
+			// 	case differenceTillNextPurchase >= 30:
+			// 		item.urgency = 'Not soon';
+			// 		break;
+			// 	default:
+			// 		item.urgency = 'Inactive';
+			// 		break;
+			// }
+			if (item.isChecked) {
+				item.urgency = 'Purchased';
+			} else if (differenceTillNextPurchase < 0) {
+				item.urgency = 'Overdue';
+			} else if (differenceTillNextPurchase <= 7) {
+				item.urgency = 'Soon';
+			} else if (
+				differenceTillNextPurchase > 7 &&
+				differenceTillNextPurchase < 30
+			) {
+				item.urgency = 'Kind of soon';
+			} else if (differenceTillNextPurchase >= 30) {
+				item.urgency = 'Not soon';
+			} else {
+				item.urgency = 'Inactive';
+			}
+		}
+	});
+
+	items.sort(
+		(firstItem, secondItem) =>
+			firstItem.isChecked - secondItem.isChecked ||
+			firstItem.isInactive - secondItem.isInactive ||
+			firstItem.days - secondItem.days ||
+			firstItem.name.localeCompare(secondItem.name),
+	);
```

Comment:
What is the point of sorting it, if you are populating the lists based on urgency?